purchase_receipt_percentage/models/purchase_order_line.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    receipt_percentage = fields.Float(compute="_compute_receipt_percentage", store=True)
+
+    @api.depends("qty_received", "product_qty")
+    def _compute_receipt_percentage(self):
+        for line in self:
+            perc = 100  # 100% by default, in case ordered qty is 0
```

Comment:
If you have any ordered qty, your receipt percentage is 0. Any reason to use 100% as default ?

Reply:
Yep, that's the idea: by default, the value is 100%, but if we the ordered qty is different than 0, then we recompute the values by dividing the received qty by the ordered qty.
Result:
- lines with 0 qty ordered => `100%` by default
- lines with X qty ordered and Y qty received =>` Y/X*100` of received percentage

Reply:
In my POV : 

- lines with 0 qty ordered: percentage received = 0% by default (because 100% means that we received anything and it's not the case)
- lines with X qty ordered and Y qty received: percentage received = Y/X*100 of received percentage